text/0031-corejam.md

```diff
@@ -0,0 +1,644 @@
+6. Utilization of Coretime must be permissionless.
+7. The nature of Coretime should closely match the nature of resources generated by Polkadot.
+8. Minimal opinionation should be introduced over the format, nature and usage of Coretime.
+
+## Stakeholders
+
+1. Anyone with exposure to the DOT token economy.
+2. Anyone wanting to create decentralised/unstoppable/resilient applications.
+3. Teams already building on Polkadot.
+
+## Explanation
+
+**CoreJam is a general model for utilization of Polkadot Cores. It is a mechanism by which Work Packages are communicated, authorized, computed and verified, and their results gathered, combined and accumulated into particular parts of the Relay-chain's state.**
+
+### From Old to New
+
+The current specification of the Polkadot protocol, and with in the Relay-chain operation, is designed in line with the overall requirements of and terminology in the Polkadot (1.0) whitepaper. It incorporates first-class concepts including *Proof-of-Validity* and *Parachain Validation Function*. This will no longer be considered canonical for the Polkadot protocol. To avoid confusion, this design will be known as the *Fixed-function Parachains Legacy Model*, or the *Legacy Model* for short.
+
+Existing functionality relied upon by parachains will continue to be provided as a special case under a more general and permissionless model which is detailed presently and known as *CoreJam*. Transition of Polkadot to be in line with the present proposal will be necessarily imply some minor alterations of formats utilized by Cumulus, Smoldot and other light-client APIs (see the section on Compatibility). However, much of the underlying logic (in particular, consensus, disputes and availability) is retained, though its application is generalised. This proposal will only make note of the expectations regarding the changes, and presumes continuation of all other logic.
+
+As part of this model, we introduce a number of new and interrelated concepts: *Work Package*, *Work Class*, *Work Item*, *Work Output*, *Work Result*, *Work Report* and *Work Package Attestation*, *Work Class Trie*.
+
+Focussing on continuity and reuse of existing logic, it is unsurprising that many of these concepts already analogues in the Parachains model, albeit ones with a less general definition. While this mapping can be helpful to quickly create an approximate understanding of the new concepts for those already familiar with Polkadot, care must be taken not to inadvertantly make incorrect presumptions over exact details of their relationships, constraints, timing, provisions and APIs. Nonetheless, they are provided here for whatever help they may be.
+
+| CoreJam model                | Legacy model      | Context |
+| --- | --- | --- |
+| *Work Package*               | Proof-of-Validity | Untrusted data provided to RcVG |
+| *Work Item*                  | Proof-of-Validity | State-transition inputs and witness |
```

Comment:
PoV was always long & unwieldy, but..  An under appreciated feature was that "work" is "verification", not "fresh computation".  It's common that computations live in the complexity classes like X ∩ NY where Y < X is the complexity of the verification problem, and considerably faster.  Sort is O(n log n) as fresh computation, but only O(n) as verification, given enough space of course.  Integer linear programs are NP-complete, but their verification is fast, and ["the integrality gap [typically] translates into the approximation ratio of an approximation algorithm"](https://en.wikipedia.org/wiki/Linear_programming_relaxation) (what you'd run on collators).  Cryptography has highly asymmetric proof and verification phases.  I'm not saying "Work" doesn't work, but we're giving up some intuition here.

Reply:
It is work from the perspective of what Polkadot is selling, even if it's not the exact same work as that which happens at some other stage of any given use-case's pipeline.